src/display/editor/drawers/inkdraw.js

```diff
@@ -0,0 +1,879 @@
+          )
+        );
+        serializedPoints.push(
+          Outline._rescale(
+            points,
+            tx,
+            ty,
+            sx,
+            sy,
+            isForCopying ? new Array(points.length) : null
+          )
+        );
+      }
+      [x1, y1, x2, y2] = [
+        tx + x1 * sx,
+        ty + y1 * sy,
+        tx + x2 * sx,
+        ty + y2 * sy,
+      ];
+    }
+
+    return {
+      lines: serializedLines,
+      points: serializedPoints,
+      rect: [x1, y2, x2, y1],
+    };
+  }
+
+  static deserialize(
+    pageX,
+    pageY,
+    pageWidth,
+    pageHeight,
+    innerMargin,
+    { paths: { lines, points }, rotation, thickness }
+  ) {
+    const newLines = [];
+    let tx, ty, sx, sy, swap;
+    switch (rotation) {
+      case 0:
+        swap = false;
+        tx = -pageX / pageWidth;
+        ty = pageY / pageHeight + 1;
+        sx = 1 / pageWidth;
+        sy = -1 / pageHeight;
+        break;
+      case 90:
+        swap = true;
+        tx = -pageY / pageHeight;
+        ty = -pageX / pageWidth;
+        sx = 1 / pageHeight;
+        sy = 1 / pageWidth;
+        break;
+      case 180:
+        swap = false;
+        tx = pageX / pageWidth + 1;
+        ty = -pageY / pageHeight;
+        sx = -1 / pageWidth;
+        sy = 1 / pageHeight;
+        break;
+      case 270:
+        swap = true;
+        tx = pageY / pageHeight + 1;
+        ty = pageX / pageWidth + 1;
+        sx = -1 / pageHeight;
+        sy = -1 / pageWidth;
+        break;
+    }
+
+    if (swap) {
+      for (let i = 0, ii = lines.length; i < ii; i++) {
+        newLines.push({
+          line: Outline._rescaleAndSwap(
+            lines[i].map(x => x ?? NaN),
+            tx,
+            ty,
+            sx,
+            sy
+          ),
+          points: Outline._rescaleAndSwap(
+            points[i].map(x => x ?? NaN),
+            tx,
+            ty,
+            sx,
+            sy
+          ),
+        });
+      }
+    } else {
+      for (let i = 0, ii = lines.length; i < ii; i++) {
+        newLines.push({
+          line: Outline._rescale(
+            lines[i].map(x => x ?? NaN),
+            tx,
+            ty,
+            sx,
+            sy
+          ),
+          points: Outline._rescale(
+            points[i].map(x => x ?? NaN),
+            tx,
+            ty,
+            sx,
+            sy
+          ),
+        });
+      }
+    }
+
+    const outlines = new InkDrawOutline();
+    outlines.build(
+      newLines,
+      pageWidth,
+      pageHeight,
+      1,
+      rotation,
+      thickness,
+      innerMargin
+    );
+
+    return outlines;
+  }
+
+  #getMarginComponents(thickness = this.#thickness) {
+    const margin = this.#innerMargin + (thickness / 2) * this.#parentScale;
+    return this.#rotation % 180 === 0
+      ? [margin / this.#parentWidth, margin / this.#parentHeight]
+      : [margin / this.#parentHeight, margin / this.#parentWidth];
+  }
+
+  #getBBoxWithNorMargin() {
```

Comment:
Nit: It looks like an `r` to many in the name.
```suggestion
  #getBBoxWithNoMargin() {
```